src/Symfony/Component/Yaml/Tests/DumperTest.php

```diff
@@ -216,6 +216,63 @@ public function testObjectSupportDisabledWithExceptions()
         $this->dumper->dump(['foo' => new A(), 'bar' => 1], 0, 0, Yaml::DUMP_EXCEPTION_ON_INVALID_TYPE);
     }
 
+    public function testDumpWithMultipleNullFlagsFormatsThrows()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('The Yaml::DUMP_NULL_AS_EMPTY and Yaml::DUMP_NULL_AS_TILDE flags cannot be used together.');
+
+        $this->dumper->dump(['foo' => 'bar'], 0, 0, Yaml::DUMP_NULL_AS_EMPTY | Yaml::DUMP_NULL_AS_TILDE);
+    }
+
+    public function testDumpNullAsEmptyInExpandedMapping()
+    {
+        $expected = "qux:\n    foo: bar\n    baz: \n";
```

Comment:
Can we ensure to not dump with trailing whitespaces so that this line would become this:

```suggestion
        $expected = "qux:\n    foo: bar\n    baz:\n";
```

Reply:
Because of how the code is designed, it's complicated. This is the line that handles adding a blank space is this one:

https://github.com/symfony/symfony/blob/cb16097fc5ecf194401d0930f69a883a1c2b2755/src/Symfony/Component/Yaml/Dumper.php#L128

We would have to check for the flag and the value to be dumped at this place, which seems very specific and not sure it would worth the extra complexity here for removing the blank space (which is still valid YAML). What do you think?